test/chunking-form/samples/synthetic-named-exports/_expected/es/main.js

```diff
@@ -0,0 +1,14 @@
+const d = {
+  fn: 42,
+  hello: 'hola'
+};
+
+var ns = d;
+
+var fn = d.fn;
+
+var foo = d.foo;
```

Comment:
There is one thing that is not optimal about the output which is that it does not allow for live-bindings because you copy the values from the default exports just once. Especially in the CommonJS use case I think this is problematic because one pattern that Rollup struggles with is properties that are added to `module.exports` at a later time, e.g. due to circular dependencies. From a high level, this would be fixable be generating this code instead:

```js
const d = {
  fn: 42,
  hello: 'hola'
};

console.log(d.fn);
console.log(d.foo);
console.log(d);
```

From a low level, this would mean that instead of injecting additional code fragments, you would need to adapt how the corresponding identifiers are rendered. This is quite possible, as a matter of fact this is how for CJS output, exported identifiers sometimes are rendered as e.g. `exports.foo` instead of `foo`.

Take a look at `Variable.ts`: What you need to do is make sure that `getBaseVariableName` and `getName´ return the right values. `getName` should return what is actually rendered, e.g. `d.foo` while `getBaseVariableName` should return the left-most identifier in a combined expression, i.e. `d` in the example (this is needed for correct variable deconflicting). It should also be taken care of that this works over several steps, i.e. if the default export is itself a reexported synthetic named export, then it should be possible that longer names are created, e.g. `ns.d.foo`.

This is also a suggestion for a test you could add:
```js
// main.js
import {foo} from './dep1.js';
console.log(foo);

// dep1.js
export {bar as default} from './dep2.js';

// dep2.js
export default {bar: {foo: 'works'}};
```

If you replace `console.log(foo)` with `assert.equal(foo, 'works')`, this would be a nice test to add to the `function` test category as well. `function` tests run the code instead of comparing with a snapshot and will make sure, no invalid code is generated even if we refactor stuff.

Another test that would be interesting is what code is generated when such an import crosses a chunk border, i.e. when the importer and the file with the synthetic named export end up in different chunks.

Reply:
Thanks for the review @lukastaegert ! I thought about the live bindings too, but since the ESM with named imports has const semantics, i thought it was not a big deal, however i do understand how this is important for the commonjs case.

Going to look at it!